### DISCUSSION TITLE
Get events of wire:loading in JavaScript

### DISCUSSION THREAD

**Discussion by martinszemitis** [livewire/livewire#2616] [1. Help]
I would like to use NProgress.js progress bar when livewire is loading.

For that I would need to get loading start and end events in javascript.

Is it possible somehow already (with some workaround or undocumented feature) or should I register feature request?

Thanks in advance for your help!

**Comment by artworkad** [comment]
```
document.addEventListener("DOMContentLoaded", () => {
    Livewire.hook('message.sent', (message, component) => {
        NProgress.start();
    })

    Livewire.hook('message.processed', (message, component) => {
        NProgress.done();
    })
});
```
Reference: https://laravel-livewire.com/docs/2.x/reference
